### PR TITLE
#3 Added config params for in and out directories. Removed deviceloca…

### PR DESCRIPTION
**Pull request by DenchikBY**
## 2.0.0

- Now its possible to set in and out directories for files.
- You can set the pattern by which to search for files.
- Generated keys can be switched to another case in generated classes.
- Removed dependency on devicelocale.
- Configs with baseLocale and maps moved from config.i18.json to build.yaml
- Generators replaced with fields for keys with static values.
- Arguments now can be wrapped with braces like ${key}.

Example of new config in build.yaml:
```yaml
targets:
  $default:
    builders:
      fast_i18n:i18nBuilder:
        options:
          directory_in: assets/locale
          directory_out: lib/i18n
          files_pattern: '.json'
          base_locale: en
          key_case: snake
          maps:
            - 'a'
            - 'b'
            - 'c.d'

### CODE REVIEW

**Comment by Tienisto** [comment]
Thanks for your contributions!
Looking good on the first glance :)
Will have a deeper look in a few hours.

File: lib/src/generator.dart [hunk 1]
```diff
@@ -49,45 +54,47 @@ void _generateHeader(StringBuffer buffer, List<I18nData> allLocales) {
...
-      '/// Translation happens during initialization of the widget (method call of t)');
+  buffer.writeln('/// Widgets using this method will not be updated after widget creation when locale changes.');
+  buffer.writeln('/// Translation happens during initialization of the widget (method call of t)');
   buffer.writeln('///');
   buffer.writeln('/// Usage:');
   buffer.writeln('/// String translated = t.someKey.anotherKey;');
-  buffer.writeln('$baseClassName get t {');
-  buffer.writeln('\treturn $mapVar[$localeVar];');
-  buffer.writeln('}');
+  buffer.writeln('$baseClassName t;');
```
Comment (Tienisto) [inline lib/src/generator.dart:90]: Maybe initialize with `_strings[_locale]` to avoid possible NPE?

File: lib/src/generator.dart [hunk 2]
```diff
@@ -195,42 +211,69 @@ void _generateHeader(StringBuffer buffer, List<I18nData> allLocales) {
...
+  String finalClassName = base
+      ? className
+      : className + locale.capitalize().replaceAll('-', '');
+
+  buffer.writeln();
+
+  if (base) {
+    buffer.writeln('class $finalClassName {');
+  } else {
+    buffer.writeln('class $finalClassName implements $className {');
```
Comment (Tienisto) [inline lib/src/generator.dart:257]: Brilliant. Now it shows an error for missing translations. I like that idea!

File: lib/src/model.dart [hunk 3]
```diff
@@ -78,23 +78,11 @@ class TextNode extends Value {
...
-      break;
-    }
-  }
-
-  return arguments;
+  return Utils
+      .argumentsRegex
+      .allMatches(content)
+      .map((e) => e.group(1))
+      .toList();
```
Comment (Tienisto) [inline lib/src/model.dart:87]: I cannot escape `hello \$name`, (would be `hello \\$name` in the json file).

File: lib/builder.dart [hunk 4]
```diff
@@ -1,63 +1,79 @@
...
 
+  String get filesPattern => options.config['files_pattern'] ?? defaultFilePattern;
+
   @override
   FutureOr<void> build(BuildStep buildStep) async {
-    if (buildStep.inputId.pathSegments.last == 'config.i18n.json' ||
-        buildStep.inputId.pathSegments.last.indexOf('_') != -1) return;
+    String directoryInPath = options.config['directory_in'];
+    String baseLocale = options.config['base_locale'];
+    String maps = options.config['maps'];
```
Comment (Tienisto) [inline lib/builder.dart:28]: `List<String> maps` instead of `String maps`?

File: lib/src/generator.dart (hunk 1 above)
Comment (DenchikBY) [reply lib/src/generator.dart]: Agree, added

File: lib/builder.dart (hunk 4 above)
Comment (DenchikBY) [reply lib/builder.dart]: Fixed

File: lib/src/model.dart (hunk 3 above)
Comment (DenchikBY) [reply lib/src/model.dart:87]: Extended regex with `([^\\]|^)\$\{?(\w+)\}?` to skip escaped sybmols

**Review by Tienisto — APPROVED** [review]

**Pull request merged**